### ISSUE TITLE
Error in export although transcription seems to be ok (everything "green")

### ISSUE THREAD

**Issue by Bacchushlg** [Audiveris/audiveris#283]
[Birdland2.zip](https://github.com/Audiveris/audiveris/files/3165259/Birdland2.zip)

when exporting this score and import to MuseScore, I get a error that this is no valid mxl-file. Ignoring the message I get a score that appears to completely mess this systems...

**Comment by hbitteur** [comment]
Image quality is not good at all, there are many poor recognized symbols.
Exporting the result to either MuseScore or Finale leads to a warning about invalid MusicXML file.

![FinaleErrorMessage](https://user-images.githubusercontent.com/22053634/63419079-6d0d1d00-c404-11e9-9626-de70ec881dcf.png)

The offending line is commented out in the following xml fragment: 
```xml
      <note default-x="114">
        <pitch>
          <step>A</step>
          <octave>3</octave>
        </pitch>
        <duration>2</duration>
        <tie type="start"></tie>
        <tie type="start"></tie>
<!--        <tie type="start"></tie>  PROBLEM IS HERE! -->
        <voice>5</voice>
        <type>quarter</type>
        <stem default-y="-21">down</stem>
        <staff>2</staff>
        <notations>
          <tied type="start" orientation="over"></tied>
          <tied type="start" orientation="under"></tied>
          <tied type="start" orientation="under"></tied>
        </notations>
      </note>
```
This relates to the A3 note located in the middle of measure 47 (middle of very last system in sheet)

![NoteWithThreeTies](https://user-images.githubusercontent.com/22053634/63419396-06d4ca00-c405-11e9-90a3-ddc3d77372c7.png)

Notice that Audiveris "detected" 3 ties between the two A3 notes. This is so because the lower eighth notes were not detected at all, plus a small staff line portion was mistaken for a slur. All in all, we end up with 3 ties applying to a single note, both on left side and on right side of the slurs.
Audiveris does not (yet) detect this error, but obviously MuseScore and Finale are more strict on this. See the "(tie,tie?)?" that appears in specification of element type "note" in Finale error message.

Manually commenting out the 3rd tie element in the xml file fixed the problem. (I could instead manually remove the additional tie slur from within Audiveris UI).

FWIW, these music editors choke of the 3rd "tie" element, but accept the 3rd "tied" element a few lines below. Don't ask me why.

I will try to update Audiveris so that no note can exhibit more then 2 ties. The question is: which ones should be kept and which ones dropped?

**Comment by Bacchushlg** [comment]
Please have a look at my appended omr-file: the described error above is no more present.
It took me half an hour to correct the errors - due to the very bad quality of the source image. But finally everything seemed to be fine.
What made me unhappy was the bad result in the xml-output file - and my question was, if you see a reason for this (and maybe also a solution how to solve it)

**Comment by hbitteur** [comment]
There were a few bugs that resulted in exceptions impeding correct MusicXML output. These are now fixed.
I pushed commit 2e98923f677513f2697565a1c8ea6da81f112ad2 to development branch.

A more puzzling problem comes from note head (id: 17311, a MANUAL inter), in measure 4 of system 1. Just enter its ID in InterBoard and you'll be pointing to this offending note.
I say "offending", because its related staff is reported to be 1 (first staff), whereas it should be 2 (second staff). Same thing for its related natural sign (reported staff 1 instead of 2).
The other note (17330), within the same chord, is correctly assigned to staff 2.
This leads to problems the software cannot fully handle, and the export of the offending note is abandonned (but, apart from this precise note, the rest of the score is exported correctly).

I manually deleted the offending note and natural sign, then reinserted them, paying attention to use the right staff.
Also in this chord, there were 4 instances of Chord-Stem relations, all between this chord and its stem. This should not occur, I will need to add a check in the software. Just in case, I deleted 3 of these relations, to keep just 1.
Anyway, after these manual corrections, the export went OK, with no error message at all.

**Comment by hbitteur** [comment]
Another problem in this score is that system 4 was wrongly detected as starting by a curly brace, resulting in a single 2-staff part.

When the score is built (for example when exporting to MusicXML), all the parts found in systems (in all pages of the score), are browsed to detect the corresponding logical parts. The heuristic is based on the number of staves per part and also the location within the system of the first 2-staff part if any.

Having this (wrong) 2-staff part in the middle of the sheet systems ruins the detection of logical parts (instead of detecting 2 logical parts, each containing 1 staff), it finds 5 different logical parts. This results in many dummy parts pushed to the MusicXML output, a dummy part being filled with measure whole rests.

I imagine a good editor like MuseScore allows you to discard these spurious parts, but it would certainly be more convenient to avoid them in Audiveris in first place. 

For this, I need to add a user feature to manually modify the system/part/staff hierarchy. 
This is something I long considered too complex to implement. But the system merge feature now works, and the manual "staff dispatching to parts" is something rather similar, so its complexity must be comparable. I need to investigate this further.

**Comment by Bacchushlg** [comment]
I have removed the curly brace which seems to leed to a reasonable system structure (the rect. brace is missing, but this is normally not critical as long as the systems are correctly detected).
But unfortunately the export still contains the curley brace. Some traces of this brace seem to resist to being deleted...

**Comment by hbitteur** [comment]
The curly brace on the left side of a system is not like a basic symbol you can easily add or remove. 
It is used during the GRID step to detect the multi-staff parts.

You can "naively" remove this brace symbol later, but unfortunately, this has (yet) no impact on the existing stave/part hierarchy. This is the feature I'll need to work upon.

Removing the brace would mean to split the 2-staff part into two separate 1-staff parts. Measures would get vertically split as well. Many features don't cross a part boundary:  slurs, beams, stems, voices... Plus the ability to undo/redo all this.
I'm not saying it's impossible, but it's certainly touchy.

A "poor man" approach could be a user option to forbid multi-staff parts in a sheet.
In fact, we already have the reverse. In BarsRetriever, the class in charge of the gathering of staves into parts, the application constant "**forceSinglePart**", when set to true, forces any 2-staff system into a single part, even if no brace is present. I did that on MuseScore advice, because some piano parts don't contain such curly braces.
So what about a constant like "**forbidSinglePart**" or perhaps "**forceSeparateParts**"?
This approach requires the user to explicitly set the constant _before_ the score is processed, but it is awfully simple.

Perhaps, we could offer both possibilities. The poor man one right now, and the full one (with all the pieces to modify) later on.

**Comment by hbitteur** [comment]
The "poor man" approach is now available in develpment branch (commit fab4cc62425a35e7436866292fd7b66e12b24c33).
When constant forceSeparateParts (in BarsRetriever class) is set to true, every staff will end up as a separate part, regardless of braces.

**Comment by Bacchushlg** [comment]
problem is solved with version 145bc93.
The user now can remove the wrongly detected piano bracket and add the correct rectangle bracket.
